src/types/tests/schema-test.ts

```diff
@@ -628,6 +628,95 @@ describe('schema', () => {
     assert.deepEqual(intersection.fields, schema3.fields);
     assert.deepEqual(intersection.refinement, schema3.refinement);
   }));
+
+  const verifyUnionOf = async (typeBuilder: (type: string) => string) => {
+    const manifest = await Manifest.parse(`
+            schema A
+              foo: ${typeBuilder(`Foo {a: Text, b: Text}`)}
+
+            schema B
+              foo: ${typeBuilder(`Foo {a: Text}`)}
+            `);
+    const aType = manifest.findSchemaByName('A');
+    const bType = manifest.findSchemaByName('B');
+
+    deleteFieldRecursively(aType, 'location', {replaceWithNulls: true});
+    deleteFieldRecursively(bType, 'location', {replaceWithNulls: true});
+
+    // Union the names
+    const unionT = aType.toLiteral();
+    unionT.names = ['A', 'B'];
+
+    assert.deepEqual(Schema.union(aType, bType).toLiteral(), unionT);
+    unionT.names = ['B', 'A'];
+    assert.deepEqual(Schema.union(bType, aType).toLiteral(), unionT);
+  };
+  const verifyIntersectOf = async (typeBuilder: (type: string) => string) => {
+    const manifest = await Manifest.parse(`
+            schema A
+              foo: ${typeBuilder(`Foo {a: Text, b: Text}`)}
```

Comment:
Ah, passing a lambda is clever! Didn't think of it. :)

Reply:
Thanks :) Seen it somewhere else (hope it isn't hard to debug)